src/grpc/packet_router/hpr_packet_down.erl

```diff
@@ -38,6 +86,33 @@ window(WindowMap) ->
         datarate = maps:get(datarate, WindowMap, Template#window_v1_pb.datarate)
     }.
 
+-spec window(non_neg_integer(), 'undefined' | non_neg_integer(), atom()) ->
+    packet_router_pb:window_v1_pb().
+window(TS, FrequencyHz, DataRate) ->
+    WindowMap = #{
+        timestamp => TS,
+        frequency => FrequencyHz,
+        datarate => DataRate
+    },
+    hpr_packet_down:window(WindowMap).
+
+-spec new_downlink(
+    Payload :: binary(),
+    Timestamp :: non_neg_integer(),
+    Frequency :: atom() | number(),
+    DataRate :: atom() | integer(),
+    Rx2 :: packet_router_pb:window_v1_pb() | undefined
+) -> downlink_packet().
+new_downlink(Payload, Timestamp, FrequencyHz, DataRate, Rx2) ->
+    PacketMap = #{
+        payload => Payload,
+        rx1 => #{
+            timestamp => Timestamp,
+            frequency => FrequencyHz,
+            datarate => DataRate
+        }
+    },
+    hpr_packet_down:to_record(PacketMap, Rx2).
 %% ------------------------------------------------------------------
 % EUnit tests
 %% ------------------------------------------------------------------
```

Comment:
Can we add some units to test all the functions?

Reply:
Will do.

Reply:
Done.